test_config.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Default TEST_CONFIG_OVERRIDE for python repos.
+
+# You can copy this file into your directory, then it will be inported from
+# the noxfile.py.
+
+# The source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/test_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    'ignored_versions': ["2.7"],
+
+    # Declare optional test sessions you want to opt-in. Currently we
+    # have the following optional test sessions:
+    #     'cloud_run' # Test session for Cloud Run application.
+    'opt_in_sessions': [],
+
+    # Only relevant for the `cloud_run` session. Specify the file
+    # names for your e2e test.
+    'cloud_run_e2e_test_files': ['e2e_test.py'],
+
+    # If set to True, the test will install the library from the root
+    # of the repository.
+    'install_library_from_source': False,
```

Comment:
I _think_ this functionality should be configurable per build - not _per location_? Sometimes we want the same project to be tested one way and sometimes another, so I think this config should only be an environment variable.

Reply:
The intent of this particular setting was to allow for the library (e.g., `google-cloud-storage`) to be installed from source `pip install -e ..` instead of from PyPI. It will be `True` in client library repos and `False` in python-docs-samples.

If you'd prefer this happen automagically I could move the detection logic into the noxfile. Searching for a `setup.py` in the repo root is probably sufficient

Reply:
Didn't we decide that samples in the client library repos would have 2 different builds: "head vs head" and "published vs published"?

My point was that those two builds are happing to the same tests (so same config), so I think this config can't be in the file (it needs to come from the build, probably via an env var).

Reply:
Ah I see your point though. I think we can yank this from the config and have the noxfile look for `INSTALL_LIBRARY_FROM_SOURCE` in the environment. 👍 

@tmatsuo, would you mind removing this?